rxbinding-recyclerview-v7/src/main/java/com/jakewharton/rxbinding/support/v7/widget/RecyclerViewItemClickOnSubscribe.java

```diff
@@ -0,0 +1,48 @@
+package com.jakewharton.rxbinding.support.v7.widget;
+
+import android.support.v7.widget.RecyclerView;
+import android.view.View;
+import com.jakewharton.rxbinding.internal.MainThreadSubscription;
+import rx.Observable;
+import rx.Subscriber;
+
+import static com.jakewharton.rxbinding.internal.Preconditions.checkUiThread;
+
+public class RecyclerViewItemClickOnSubscribe implements Observable.OnSubscribe<Integer> {
+
+  private final RecyclerView recyclerView;
+
+  public RecyclerViewItemClickOnSubscribe(RecyclerView recyclerView) {
+    this.recyclerView = recyclerView;
+  }
+
+  @Override public void call(final Subscriber<? super Integer> subscriber) {
+    checkUiThread();
+
+    final RecyclerView.OnChildAttachStateChangeListener childAttachStateChangeListener =
+        new RecyclerView.OnChildAttachStateChangeListener() {
+          @Override public void onChildViewAttachedToWindow(View view) {
+            recyclerView.getChildViewHolder(view).itemView.setOnClickListener(
```

Comment:
To me the library API is clear enough: RxRecyclerView.itemClicks indicates item(in this case, the whole ViewHolder) click events. If user wants something different, they should use RxView.clicks.